run_xnat2bids.py

```diff
@@ -11,65 +13,115 @@
 
 
 logging.basicConfig(format='%(levelname)s: %(message)s', level=logging.NOTSET)
+logging.getLogger('asyncio').setLevel(logging.WARNING)
+
+# Define a coroutine function to run subprocess command
+async def run_subprocess(srun_cmd):
+    await asyncio.create_subprocess_exec(*srun_cmd)
+   
+
 
 def extractParams(param, value):
     arg = []
     for v in value:
         arg.append(f"--{param} {v}")
     return ' '.join(arg)
 
-def compileArgumentList(arg_dict, user):
+def mergeConfigFiles(user_cfg, default_cfg):
+
+        user_slurm = user_cfg['slurm-args']
+        user_x2b = user_cfg['xnat2bids-args']
+        default_slurm = default_cfg['slurm-args']
+        default_x2b = default_cfg['xnat2bids-args']
+
+        # Update default config with user provided parameters
+        default_slurm.update(user_slurm)
+        default_x2b.update(user_x2b)
+
+        # Assemble final argument list 
+        merged_dict = {}
+        merged_dict['slurm-args'] = default_slurm
+        merged_dict['xnat2bids-args'] = default_x2b
+
+        # Add session specific parameter blocks
+        for key in user_cfg.keys():
+            if key == 'slurm-args' or key == 'xnat2bids-args':
+                continue
+            merged_dict[key] = user_cfg[key]
+
+        return merged_dict
+
+def compileX2BArgList(xnat2bids_dict, session):
+        x2b_param_list = []
+        param_lists = ["includeseq", "skipseq"]
+        
+        for param, value in xnat2bids_dict.items():
+            if value != "" and value is not None:
```

Comment:
Instead of indenting this whole block with the `value` stuff, try reversing the logic and adding a continue. Then you can unindent the rest of the block

```python
if value == "" or value is None:
    arg = f"--{param} {value}"
    x2b_param_list.append(arg)
    continue
```

Make sure I got that conditional right! When I'm writing code like this I try my hardest to indent the least amount. that way I don't need to keep track of which `if` I'm in, or if I'm in the `else`. It's mostly a style thing, but it's something that has treated me well in the past